src/rdiff_backup/Main.py

```diff
@@ -74,7 +74,7 @@ def _main_run(arglist, security_override=False):
 
     # validate that everything looks good before really starting
     ret_val = action.pre_check()
-    if ret_val != 0:
+    if ret_val & Globals.RET_CODE_ERR:
```

Comment:
If my understanding is right, the process will continue running is pre_check, or check return RET_CODE_FILE ?

What is the rational behind this ?

Reply:
As I wrote, it's not fully implemented but the idea is that only a "real" error stops the program workflow, a warning or a single file error (especially at backup time) doesn't stop the whole backup.

IMHO it's not a change of logic: already today, the backup doesn't stop just because a few files can't be saved (e.g. because not readable).
I only formalize the approach with predefined return codes.